File: LearningManagementBackend/src/main/java/org/learn/learningmanagementbackend/model/AcademicYear.java
package org.learn.learningmanagementbackend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.Fetch;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "Academic_year")
@NoArgsConstructor
@AllArgsConstructor
public class AcademicYear extends BaseEntity{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "school_id")
    private School school;

    @Column(name = "name", length = 20)
    private String name;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Column(name = "is_active")
    private Boolean isActive;

    @OneToMany(mappedBy = "academicYear", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Semester> semesters = new ArrayList<>();
}

File: learning_management_app/lib/models/admin/academic_year.dart
class AcademicYear {
  final int? id;
  final int? schoolId;
  final String? name;
  final String? startDate;
  final String? endDate;
  final bool? isActive;

  AcademicYear({
    this.id,
    this.schoolId,
    this.name,
    this.startDate,
    this.endDate,
    this.isActive,
  });

  factory AcademicYear.fromJson(Map<String, dynamic> json) {
    return AcademicYear(
      id: json['id'],
      schoolId: json['schoolId'],
      name: json['name'],
      startDate: json['startDate'],
      endDate: json['endDate'],
      isActive: json['isActive'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'schoolId': schoolId,
      'name': name,
      'startDate': startDate,
      'endDate': endDate,
      'isActive': isActive,
    };
  }
}
